### PR TITLE
Split hatchlings and eggs, re-sort ordering

### PR DESCRIPTION
**Pull request by danblackadder**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,3 +26,19 @@ docker compose exec hatchery bash -c "npm run db:migrate"
...
+
+```bash
+npm run dev:docker:cleanup
+```
+
+If you need to remove the database and other volumes you can run:
+
+```bash
+npm run dev:docker:cleanup:all
+```
```
Comment (edenchazard) [inline README.md:42]: **legend**!

File: README.md [hunk 2]
```diff
@@ -26,3 +26,19 @@ docker compose exec hatchery bash -c "npm run db:migrate"
 - [Tailwind](https://tailwindcss.com/)
 - [Nuxt](https://nuxt.com/)
 - [Vue.js](https://vuejs.org/)
+
+## Troubleshooting
+
+If you need to start from scratch you can run the following commands to remove the docker containers and images whilst keeping the database intact:
```
Comment (edenchazard) [inline README.md:32]:
```suggestion
If you need to start from scratch, you can run the following commands to remove the docker containers and images whilst keeping the database intact:
```

File: global.d.ts [hunk 3]
```diff
@@ -27,6 +27,8 @@ declare global {
   type ScrollView = DragonData & {
     in_garden: boolean;
     in_seed_tray: boolean;
+    incubated?: boolean;
+    stunned?: boolean;
```
Comment (edenchazard) [inline global.d.ts:31]: I'm guessing these being optional is because only either one or the other can be present, yes?

File: pages/index.vue [hunk 4]
```diff
@@ -85,27 +85,54 @@
...
-            :key="dragon.id"
-            v-model="scroll.dragons[i]"
-            :settings="userSettings"
-            :recently-added
-            @click="
-              () => {
-                if (!isProcessing) {
-                  dragon.in_garden = !dragon.in_garden;
+        <template v-if="hatchlings.length">
+          <p>Hatchlings</p>
```
Comment (edenchazard) [inline pages/index.vue:89]: probably wants to be a header tag

Comment (edenchazard) [inline pages/index.vue:89]: think the text as well should be smaller here. just concerned with it becoming a lengthy page

File: pages/index.vue [hunk 5]
```diff
@@ -85,27 +85,54 @@
...
+                  if (!isProcessing) {
+                    hatchling.in_garden = !hatchling.in_garden;
+                  }
+                }
+              "
+            />
+          </div>
+        </template>
+        <template v-if="eggs.length">
+          <p>Eggs</p>
```
Comment (edenchazard) [inline pages/index.vue:113]: h

File: server/api/user/scroll/index.ts [hunk 6]
```diff
@@ -103,16 +104,23 @@ export default defineEventHandler(async (event) => {
...
+        dragon.stunned = isStunned(dragon);
 
-      return {
-        ...scrollResponse.dragons[id],
-        in_garden: !!(hatcheryDragon?.in_garden ?? false),
-        in_seed_tray: !!(hatcheryDragon?.in_seed_tray ?? false),
-      };
-    }),
+        return dragon;
+      }),
```
Comment (edenchazard) [inline server/api/user/scroll/index.ts:124]: I mean, this obviously works and isn't necessarily wrong as such, but you're mutating the value and returning it when map is really about returning a new value, no?

**Review by edenchazard — CHANGES_REQUESTED** [review]

File: global.d.ts (hunk 3 above)
Comment (danblackadder) [reply global.d.ts:31]: Exactly

File: server/api/user/scroll/index.ts (hunk 6 above)
Comment (danblackadder) [reply server/api/user/scroll/index.ts]: 🤷

Comment (danblackadder) [reply server/api/user/scroll/index.ts]: hold my beer

**Review by edenchazard — APPROVED** [review]

**Pull request merged**